inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -285,3 +295,47 @@ def refextract(obj, eng):
     elif len(text_references) > len(pdf_references):
         obj.log.info('Extracted %d references from text.', len(text_references))
         obj.data['references'] = text_references
+
+
+@with_debug_logging
+def set_document_type_and_refereed(obj, eng):
+    publications = obj.data.get('publication_info')
+    journals = filter(lambda x: 'journal_title' in x, publications)
+
+    modify_doc_type = False
+    modify_ref_f = False
+
+    if not journals:
+        return None
+
+    for index, journal in enumerate(journals):
+        query = RecordMetadata.query.filter(
```

Comment:
you don't need to do all this. In principle, after #2938, the `journal_record` will have been written in the `publication_info`, so you can just get the record from the DB instead of doing this complicated query.

Reply:
I am not sure how i can get it with just the reference?